src/components/connection_handler/src/connection_handler_impl.cc

```diff
@@ -1688,9 +1703,11 @@ void ConnectionHandlerImpl::KeepConnectionAlive(uint32_t connection_key,
 }
 
 void ConnectionHandlerImpl::OnConnectionEnded(
-    const transport_manager::ConnectionUID connection_id) {
+    const transport_manager::ConnectionUID connection_id,
+    const CloseSessionReason close_reason) {
   SDL_LOG_INFO("Delete Connection: " << static_cast<int32_t>(connection_id)
-                                     << " from the list.");
+                                     << " from the list."
+                                     << " with reason " << close_reason);
```

Comment:
It seems to me that the `reason` could be determined here rather than passed in.

something like "If the final message was sent, send use kFinalMessage, otherwise use kCommon"

Reply:
@jacobkeeler 
Yes, of course, but CloseSessionReason is an OnConnectionEnded parameter. 
We call this function in several places.
I believe that the separation of the logic of removing the connection and finding out the reason is justified in this case.

Reply:
I still think it would be better to have this logic within the function, since `unexpected_disconnect` should always be `false` if the last message was sent, plus it is better to have the logic in one place.

Reply:
@jacobkeeler, I agree that extending arguments of `OnConnectionEnded` note required. 
Moved `ClosureReasonHandling` into `OnConnectionEnded` with 21fcf233f3